portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -28,29 +39,81 @@
+   * Blobstore. The POST request also results in a re-direct back to the original server-dev page.
+   * TODO(Issue #15): Do verfification on a new comment before adding it to the comments list.
+   *
+   * @param request The <code>HttpServletRequest</code> for the POST request.
+   * @param response The <code>HttpServletResponse</code> for the POST request.
+   * @return void. A Entity of the Comment kind is created and upserted to the Datastore.
+   */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String newComment = request.getParameter("comment");
     long timestamp = System.currentTimeMillis();
+    String imageUrl = getUploadedFileUrl(request, "image");
 
     Entity taskEntity = new Entity("Comment");
     taskEntity.setProperty("text", newComment);
     taskEntity.setProperty("timestamp", timestamp);
-
+    taskEntity.setProperty("imageUrl", timestamp);
     datastore.put(taskEntity);
 
     response.sendRedirect("/pages/server-dev.html");
   }
+
+  /**
+   * Returns a String corresponding to the URL that points to the uploaded file, or null if the user
+   * didn't upload a file.
+   *
+   * @param request The <code>HttpServletRequest</code> for the POST request.
+   * @param formInputElementNameThe Name attribute of the image file input to the form.
+   * @return The URL that points to the uploaded file.
+   */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Check the validity of the file here by making sure it's an image file
+    if (!"image".equals(blobInfo.getContentType().substring(0, 5))) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // Use imagesService to get a URL that points to the uploaded file.
+    ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+
+    // To support running in Google Cloud Shell with AppEngine's devserver, we must use the relative
+    // path to the image, rather than the path returned by imagesService which contains a host.
+    try {
+      URL url = new URL(imagesService.getServingUrl(options));
+      return url.getPath();
+    } catch (MalformedURLException e) {
+      return imagesService.getServingUrl(options);
+    }
```

Comment:
Can you please modify this code to avoid calling imagesService.getServingUrl(options) twice?

Reply:
Fixed.